python/lsst/ip/isr/isrTask.py

```diff
@@ -574,16 +713,117 @@ class IsrTask(pipeBase.CmdLineTask):
         self.makeSubtask("crosstalk")
         self.makeSubtask("strayLight")
         self.makeSubtask("fringe")
         self.makeSubtask("masking")
         self.makeSubtask("vignette")
 
+    @classmethod
+    def getInputDatasetTypes(cls, config):
+        inputTypeDict = super().getInputDatasetTypes(config)
+        obsConfigDir = os.path.join(getPackageDir("obs_subaru"), "config", "hsc")
+        config.load(os.path.join(obsConfigDir, "isr.py"))
+
+        # Delete entries from the dictionary of InputDatasetTypes that we know we don't
+        # need because the configuration tells us we will not be bothering with the
+        # correction that uses that IDT.
+        if config.doBias is not True:
+            del inputTypeDict["bias"]
+        if config.doLinearize is not True:
+            del inputTypeDict["linearizer"]
+        if config.doCrosstalk is not True:
+            del inputTypeDict["crosstalkSources"]
+        if config.doBrighterFatter is not True:
+            del inputTypeDict["bfKernel"]
+        if config.doDefect is not True:
+            del inputTypeDict["defects"]
+        if config.doDark is not True:
+            del inputTypeDict["dark"]
+        if config.doFlat is not True:
+            del inputTypeDict["flat"]
+        if config.doFringe is not True:
+            del inputTypeDict["fringes"]
+        if config.doAttachTransmissionCurve is not True:
+            del inputTypeDict["opticsTransmission"]
+            del inputTypeDict["filterTransmission"]
+            del inputTypeDict["sensorTransmission"]
+            del inputTypeDict["atmosphereTransmission"]
+        if config.doUseOpticsTransmission is not True:
+            del inputTypeDict["opticsTransmission"]
+        if config.doUseFilterTransmission is not True:
+            del inputTypeDict["filterTransmission"]
+        if config.doUseSensorTransmission is not True:
+            del inputTypeDict["sensorTransmission"]
+        if config.doUseAtmosphereTransmission is not True:
+            del inputTypeDict["atmosphereTransmission"]
+
+        return inputTypeDict
+
+    @classmethod
+    def getOutputDatasetTypes(cls, config):
+        outputTypeDict = super().getOutputDatasetTypes(config)
+        obsConfigDir = os.path.join(getPackageDir("obs_subaru"), "config", "hsc")
+        config.load(os.path.join(obsConfigDir, "isr.py"))
+
+        if config.qa.doThumbnailOss is not True:
+            del outputTypeDict["outputOssThumbnail"]
+        if config.qa.doThumbnailFlattened is not True:
+            del outputTypeDict["outputFlattenedThumbnail"]
+
+        return outputTypeDict
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
+        try:
+            inputData['detectorNum'] = int(inputDataIds['ccdExposure']['detector'])
+        except Exception as e:
+            raise ValueError("Failure to find valid detectorNum value for Dataset {}: {}".format(
+                inputDataIds, e))
```

Comment:
I just learned about `f""` formatted strings and they are really compact.  See a description [here](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python).  I'm not requiring you change, but Tim introduced them to me and I thought they were so much better than previous formatting that I wanted to spread the word.